lib/src/working_copy.rs

```diff
@@ -644,7 +644,7 @@ impl TreeState {
 
         let sparse_matcher = self.sparse_matcher();
 
-        let fsmonitor_clock_needs_save = fsmonitor_kind.is_some();
+        let mut is_dirty = fsmonitor_kind.is_some();
```

Comment:
It's no longer obvious that the reason it's dirty is that the fsmonitor clock needs to be saved, while the rest of the updates to `is_dirty` are because the tree state/working copy state has changed. (I personally don't conceptualize the fsmonitor clock changing as necessarily having "dirtied" anything.) Maybe leave the intermediate variable name? Up to your discretion.

```rust
let fsmonitor_clock_needs_save = fsmonitor_kind.is_some();
let mut is_dirty = fsmonitor_clock_needs_save;
```

Reply:
Yes, makes sense. I've restored the variable. An alternative would be to do `let mut is_dirty = watchman_clock != self.watchman_clock;`, but I suppose it's practically always going to be different anyway, unless it's `None`.